message/message-util/util/src/java/org/sakaiproject/message/util/BaseMessage.java

```diff
@@ -1724,6 +1716,25 @@ public String merge(String siteId, Element root, String archivePath, String from
 				channel = edit;
 			}
 
+			// Get the existing titles for duplicate removal
+			Set<String> messageTitles = new LinkedHashSet<>();
+			try
+			{
+				Iterator messages = channel.getMessages(null, true).iterator();
```

Comment:
Use a for instead of iterator + while

Reply:
Thanks.

Reply:
I will use streams :)

Reply:
I tried to use streams - it did no go well.  Since the getMessages() returns an Iterator, the way to use a for look is to turn the Iterator into a List and look through that.  For small numbers of messages it is equivalent using a little more memory - but for large numbers of messages (unlikely) the Iterator pattern is more elegant.  Even if under the covers the service implementation is not particularly elegant :)

Reply:
getMessages return a List

`public List getMessages(Filter filter, boolean ascending) throws PermissionException
		{
			// check security on the channel (throws if not permitted)
			unlock(SECURE_READ, getReference());
			// track event
			// m_eventTrackingService.post(m_eventTrackingService.newEvent(eventId(SECURE_READ), getReference(), false));

			return findFilterMessages(filter, ascending);

		} // getMessages`

Reply:
You can use a for

```
 List<MessageEdit> messages = channel.getMessages(null, true);
    for (MessageEdit msg : messages) {
 ```